docs/docs/api/LayoutAnimations/CustomAnimations.md

```diff
@@ -0,0 +1,230 @@
+    * `values.height` -view's height (after)
+    * `values.globalOriginX` - X coordinate of top left corder in global coordinate system (after)
+    * `values.globalOriginY` - Y coordinate of top left corder in global coordinate system (after)
+    * `values.boriginX` - X coordinate of top left corder in parent's coordinate system (before)
+    * `values.boriginY` - Y coordinate of top left corder in parent's coordinate system (before)
+    * `values.bwidth` - view's width (before)
+    * `values.bheight` -view's height (before)
+    * `values.bglobalOriginX` - X coordinate of top left corder in global coordinate system (before)
+    * `values.bglobalOriginY` - Y coordinate of top left corder in global coordinate system (before)
+
+### Example
+<video src="https://user-images.githubusercontent.com/12784455/120450759-09fa3980-c391-11eb-9b64-65ec8e6c2509.mp4" controls="controls" muted="muted" width="45%"></video>
+
+```js
+
+function CustomLayoutTransition(values) {
+  'worklet'
+  return {
+    animations: {
+      originX: withTiming(values.originX, {duration: 1000}),
+      originY: withDelay(1000, withTiming(values.originY, {duration: 1000})),
+      width: withSpring(values.width),
+      height: withSpring(values.height),
+    },
+    initialValues: {
+      originX: values.boriginX,
+      originY: values.boriginY,
+      width: values.bwidth,
+      height: values.bheight,
+    }
+  };
+}
+
+function Box({label, state}: {label: string, state: boolean}) {
+  const ind = label.charCodeAt(0) - ('A').charCodeAt(0);
+  const delay = 300 * ind;
+  return (
+    <Animated.View 
+        layout={CustomLayoutTransition} 
+        style={[styles.box,
+          { 
+            flexDirection: (state)? 'row': 'row-reverse', 
+            height: (state)? 30: 60,
+          }]} 
+    >
+      <Text> {label} </Text>
+    </Animated.View>
+  );
+}
+
+export function CustomLayoutTransitionExample(): React.ReactElement {
+  const [state, setState] =  useState(true);
+  return (
+    <View style={{marginTop: 30}} >
+      <View style={{height: 300}} >
+        <AnimatedLayout style={{flexDirection: state? 'row' : 'column'}} >
+          {state && <Box key="a" label="A" state={state} />}
+          <Box key="b" label="B" state={state} />
+          {!state && <Box key="a" label="A" state={state} />}
+          <Box key="c" label="C" state={state} />
+        </AnimatedLayout>
+      </View>
+    
+      <Button onPress={() => {setState(!state)}} title="toggle" />
+    </View>
+  );
+}
+
+
+```
+
+## Other Facts
+
+Each Reanimated component has its shared value that keeps current animations assigned to that particular component. If you want to start a new animation for a specific prop and you wouldn't provide an initial value for the prop then the initial value will be taken from the last animation that has been assigned to the component. The only exception is Entering animation because we have no way to get the previous animation values.
```

Comment:
```suggestion
Each Reanimated component has its shared value that keeps current animations assigned to that particular component. If you want to start a new animation for a specific prop and you don't provide an initial value for the prop then the initial value will be taken from the last animation that has been assigned to the component. The only exception is Entering animation because we have no way to get the previous animation values.
```